ShareExtension/CustomShareExtension/CustomShareViewController.swift

```diff
@@ -10,27 +10,40 @@ import SwiftUI
 import MobileCoreServices
 
 class CustomShareViewController: UIViewController {
+
     override func viewDidLoad() {
         super.viewDidLoad()
 
-        // 1: Set the background and call the function to create the navigation bar
-        self.view.backgroundColor = .systemGray6
+        // Try to get input items
+        if let extensionItems = extensionContext?.inputItems,
+           let attachments = (extensionItems[0] as? NSExtensionItem)?.attachments {
+            let urlProvider = attachments[0] as NSItemProvider
+
+            urlProvider.loadItem(forTypeIdentifier: "public.url", options: nil) { [weak self] (decoder: NSSecureCoding!, _) -> Void in
+                if let url = decoder as? NSURL,
+                   let imageData = NSData(contentsOf: url as URL),
+                   let uiimg = UIImage(data: imageData as Data) {
+                    DispatchQueue.main.async { [weak self] in
+                        self?.setupViews(image: uiimg)
+                    }
+                }
+            }
+        }
+
+        view.backgroundColor = .systemGray6
         setupNavBar()
-        setupViews()
     }
 
-    // 2: Set the title and the navigation items
     private func setupNavBar() {
-        self.navigationItem.title = "Custom Share Extension"
+        navigationItem.title = "Custom Share Extension"
```

Comment:
Extract all of the hardcoded strings in the code and put it in one file and reuse it from there. I.e. put all of the strings in one Strings.swift file and use it across the project.